sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract features across (shifted) windows from provided series."""
+
+__author__ = ["danbartl"]
+__all__ = ["WindowSummarizer"]
+
+import pandas as pd
+from joblib import Parallel, delayed
+
+from sktime.transformations.base import BaseTransformer
+
+
+class WindowSummarizer(BaseTransformer):
+    """Transformer for extracting time series features.
+
+    The WindowSummarizer transforms input series to features
+    based on a provided dictionary of window summarizer, window shifts
+    and window lengths.
+
+    Parameters
+    ----------
+    Dictionary with the following arguments:
+    index : str, base string of the derived features
+            final name will also contain
+            window shift and window length.
+    summarizer: either str corresponding to implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+            or function definition
+    window: list of integers
+        Contains values for window shift and window length.
+    target_cols: list of str,
+        Specifies which columns in X to target. If set to None will
+        target first column in X.
```

Comment:
I am confused, you say a dict, but we have the params
```
        summarizer=None,
        n_jobs=-1,
        target_cols=None,
```
So its not a dict actually? What you are doing in examples is just gving those 3 params as a kwargs dict.
```suggestion
    index : str, base string of the derived features
            final name will also contain
            window shift and window length.
    summarizer: either str corresponding to implemented pandas function, currently
            * "sum",
            * "mean",
            * "median",
            * "std",
            * "var",
            * "kurt",
            * "min",
            * "max",
            * "corr",
            * "cov",
            * "skew",
            * "sem"
            or function definition
    window: list of integers
        Contains values for window shift and window length.
    target_cols: list of str,
        Specifies which columns in X to target. If set to None will
        target first column in X.
```